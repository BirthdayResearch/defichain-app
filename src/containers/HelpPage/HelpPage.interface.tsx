export interface HelpPageProps {}
export interface HelpPageState {}
