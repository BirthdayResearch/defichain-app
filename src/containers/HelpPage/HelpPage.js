import React, { Component } from 'react';
import { Helmet } from "react-helmet";

class HelpPage extends Component {
  render() {
    return (
      <div className="main-wrapper">
        <Helmet>
          <title>Help â€“ DeFi Blockchain Client</title>
        </Helmet>
        <header className="header-bar">
          <h1>Help</h1>
        </header>
        <div className="content">
          <section>
            Section
          </section>
        </div>
      </div>
    );
  }
}

export default HelpPage;