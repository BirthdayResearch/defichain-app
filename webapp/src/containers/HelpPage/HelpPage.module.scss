// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../scss/variables';
@import '../../scss/bootstrap/variables';

.helpOptions {
  margin-top: 2rem;
  margin-bottom: 1rem;
}

.helpOption {
  margin-bottom: 1.5rem;
  cursor: pointer;
}

.helpOptionRow {
  display: flex;
  align-items: flex-start;
  padding: 0.75rem 1.25rem;
}

.helpOptionIcon {
  width: 48px;
  height: 48px;
  margin-right: 1rem;

  svg {
    width: 48px;
    height: 48px;
    color: $primary;
  }
}

.wechatIcon {
  width: 150px;
  height: 150px;
  margin-right: 0;
}

.helpOptionDescription {
  h3 {
    font-weight: $font-weight-normal;
    line-height: 1.5rem;
    font-size: $font-size-base;
    margin-bottom: 0;
  }

  p {
    font-size: $font-size-sm;
    color: $secondary;
    margin-bottom: 0;
    line-height: 1.25rem;
  }
}

.footerNote {
  text-align: center;
  font-size: $font-size-sm;
  color: $secondary;
}
