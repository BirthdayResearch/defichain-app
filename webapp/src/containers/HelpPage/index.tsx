import React, { useState } from 'react';
import { Helmet } from 'react-helmet';
import { FaTelegram } from 'react-icons/fa';
import { GoMarkGithub } from 'react-icons/go';
import { FaReddit } from 'react-icons/fa';
import { MdBook, MdHelp, MdSearch } from 'react-icons/md';
import { I18n } from 'react-redux-i18n';
import Header from '../HeaderComponent';
import { RouteComponentProps } from 'react-router-dom';
import { Row, Col, Card, CardBody, Popover, PopoverBody } from 'reactstrap';
import openNewTab from '../../utils/openNewTab';
import styles from './HelpPage.module.scss';
import {
  TELEGRAM_GERMAN_HELP_LINK,
  TELEGRAM_ENGLISH_HELP_LINK,
  GITHUB_ISSUE_HELP_LINK,
  DEFICHAIN_FAQ_HELP_LINK,
  getSiteURL,
  REDDIT_HELP_LINK,
  COMMUNITY_WIKI_LINK,
  CHINESE_TRADITIONAL,
  CHINESE_SIMPLIFIED,
  TELEGRAM_ZH_HELP_LINK,
  WECHAT_ZH_LINK,
  DEFICHAIN_SCAN,
} from '../../constants';
import Logo from '../../components/Svg/DefiLogo';
import { getPageTitle } from '../../utils/utility';
import { useSelector } from 'react-redux';
import { RiWechatFill } from 'react-icons/ri';
import QrCodeWeChat from '../../assets/svg/qrcode_wechat.svg';

const HelpPage: React.FunctionComponent<RouteComponentProps> = () => {
  const [popoverOpen, setPopoverOpen] = useState(false);
  const toggle = () => {
    setPopoverOpen(!popoverOpen);
  };
  const { locale } = useSelector((state: any) => state.i18n);
  const isChinese = [CHINESE_SIMPLIFIED, CHINESE_TRADITIONAL].includes(locale);
  const onLinkClick = (link: string) => {
    setPopoverOpen(false);
    openNewTab(link);
  };

  return (
    <div className='main-wrapper'>
      <Helmet>
        <title>{getPageTitle(I18n.t('containers.helpPage.title'))}</title>
      </Helmet>
      <Header>
        <h1>{I18n.t('containers.helpPage.help')}</h1>
      </Header>
      <div className='content'>
        <section>
          <p>{I18n.t('containers.helpPage.helpPageHeader')}</p>
          <Row className={styles.helpOptions}>
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(TELEGRAM_ENGLISH_HELP_LINK)}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <FaTelegram />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.telegramEn')}</h3>
                    <p>
                      {new URL(TELEGRAM_ENGLISH_HELP_LINK).hostname.replace(
                        'www.',
                        ''
                      )}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() =>
                  onLinkClick(
                    isChinese
                      ? TELEGRAM_ZH_HELP_LINK
                      : TELEGRAM_GERMAN_HELP_LINK
                  )
                }
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <FaTelegram />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>
                      {I18n.t(
                        isChinese
                          ? 'containers.helpPage.telegramZh'
                          : 'containers.helpPage.telegramDe'
                      )}
                    </h3>
                    <p>
                      {new URL(
                        isChinese
                          ? TELEGRAM_ZH_HELP_LINK
                          : TELEGRAM_GERMAN_HELP_LINK
                      ).hostname.replace('www.', '')}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            {isChinese && (
              <Col md='6'>
                <Card id='wechat_popover' className={styles.helpOption}>
                  <CardBody className={styles.helpOptionRow}>
                    <div className={styles.helpOptionIcon}>
                      <RiWechatFill />
                    </div>
                    <div className={styles.helpOptionDescription}>
                      <h3>{I18n.t('containers.helpPage.wechat')}</h3>
                      <p>
                        {new URL(WECHAT_ZH_LINK).hostname.replace('www.', '')}
                      </p>
                    </div>
                  </CardBody>
                </Card>
                <Popover
                  placement='top'
                  isOpen={popoverOpen}
                  target='wechat_popover'
                  toggle={toggle}
                >
                  <PopoverBody>
                    <img className={styles.wechatIcon} src={QrCodeWeChat} />
                  </PopoverBody>
                </Popover>
              </Col>
            )}
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(GITHUB_ISSUE_HELP_LINK)}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <GoMarkGithub />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.github')}</h3>
                    <p>
                      {new URL(GITHUB_ISSUE_HELP_LINK).hostname.replace(
                        'www.',
                        ''
                      )}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() =>
                  onLinkClick(`${getSiteURL()}${DEFICHAIN_FAQ_HELP_LINK}`)
                }
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <MdHelp />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.faq')}</h3>
                    <p>
                      {new URL(
                        `${getSiteURL()}${DEFICHAIN_FAQ_HELP_LINK}`
                      ).hostname.replace('www.', '')}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(getSiteURL())}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <Logo />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.defichainsite')}</h3>
                    <p>{new URL(getSiteURL()).hostname.replace('www.', '')}</p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(REDDIT_HELP_LINK)}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <FaReddit />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.reddit')}</h3>
                    <p>
                      {new URL(REDDIT_HELP_LINK).hostname.replace('www.', '')}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(COMMUNITY_WIKI_LINK)}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <MdBook />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.wiki')}</h3>
                    <p>
                      {new URL(COMMUNITY_WIKI_LINK).hostname.replace(
                        'www.',
                        ''
                      )}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
            <Col md='6'>
              <Card
                onClick={() => onLinkClick(DEFICHAIN_SCAN)}
                className={styles.helpOption}
              >
                <CardBody className={styles.helpOptionRow}>
                  <div className={styles.helpOptionIcon}>
                    <MdSearch />
                  </div>
                  <div className={styles.helpOptionDescription}>
                    <h3>{I18n.t('containers.helpPage.defiscan')}</h3>
                    <p>
                      {new URL(DEFICHAIN_SCAN).hostname.replace('www.', '')}
                    </p>
                  </div>
                </CardBody>
              </Card>
            </Col>
          </Row>
          <p className={styles.footerNote}>
            {I18n.t('containers.helpPage.helpTextFooterLine1')}
            <br />
            {I18n.t('containers.helpPage.helpTextFooterLine2')}
          </p>
        </section>
      </div>
    </div>
  );
};

export default HelpPage;
