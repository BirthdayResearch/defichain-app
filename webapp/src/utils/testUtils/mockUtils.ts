import PersistentStore from '../persistentStore';
import * as isElectron from '../isElectron';
import * as Notifications from '../notifications';
import log from 'loglevel';
import { runSaga } from 'redux-saga';
import axios from 'axios';

const mockPersistentStore = (getData, setData) => {
  PersistentStore.get = jest.fn(() => getData);
  PersistentStore.set = jest.fn(() => setData);
  return PersistentStore;
};

const mockAxios = mockFunc => {
  axios.create = jest.fn().mockImplementation(() => ({
    post: mockFunc,
  }));
};
const dispatchedFunc = async (method, payload = {}) => {
  const dispatched = Array();
  let action = {};
  if (payload) action = { payload };
  await runSaga(
    {
      dispatch: action => dispatched.push(action),
    },
    method,
    action
  );
  return dispatched;
};

const mockisElectron = () => {
  return jest.spyOn(isElectron, 'isElectron');
};
const mockIpcRenderer = () => {
  return jest.spyOn(isElectron, 'ipcRendererFunc');
};

const spyLogger = () => {
  return jest.spyOn(log, 'error');
};

const mockNotification = () => {
  return (
    jest
      .spyOn(Notifications, 'default')
      // tslint:disable-next-line: no-empty
      .mockImplementation(() => {})
  );
};
export {
  mockPersistentStore,
  mockAxios,
  dispatchedFunc,
  mockisElectron,
  mockIpcRenderer,
  spyLogger,
  mockNotification,
};
