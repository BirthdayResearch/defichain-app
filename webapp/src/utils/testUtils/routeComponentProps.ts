import { createMemoryHistory, createLocation } from 'history';
import { match } from 'react-router';
const history = createMemoryHistory();
const path = `/route/:id`;
const match: match<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(':id', '1'),
  params: { id: '1' },
};

const location = createLocation(match.url);
delete history.location.key;
delete location.key;

export { history, match, location };
