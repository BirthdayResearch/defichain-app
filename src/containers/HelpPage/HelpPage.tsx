import React, { Component } from "react";
import { Helmet } from "react-helmet";
import { HelpPageProps, HelpPageState } from "./HelpPage.interface";

class HelpPage extends Component<HelpPageProps, HelpPageState> {
  render() {
    return (
      <div className="main-wrapper">
        <Helmet>
          <title>Help â€“ DeFi Blockchain Client</title>
        </Helmet>
        <header className="header-bar">
          <h1>Help</h1>
        </header>
        <div className="content">
          <section>Section</section>
        </div>
      </div>
    );
  }
}

export default HelpPage;
