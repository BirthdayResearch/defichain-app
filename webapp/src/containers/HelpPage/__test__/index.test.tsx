import React from 'react';
import { shallow } from 'enzyme';
import HelpPage from '../index';
import * as routeComponentProps from '../../../utils/testUtils/routeComponentProps';

describe('HelpPage Page', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<HelpPage {...routeComponentProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
