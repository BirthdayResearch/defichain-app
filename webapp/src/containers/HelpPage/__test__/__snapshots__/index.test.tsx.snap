// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage Page should check for snapshot 1`] = `
<div
  className="main-wrapper"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Help - DeFi Wallet
    </title>
  </HelmetWrapper>
  <Connect(Header)>
    <h1>
      Help
    </h1>
  </Connect(Header)>
  <div
    className="content"
  >
    <section>
      <p>
        Connect with the community with these options:
      </p>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <FaTelegram />
              </div>
              <div>
                <h3>
                  Telegram community (EN)
                </h3>
                <p>
                  t.me
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <FaTelegram />
              </div>
              <div>
                <h3>
                  Telegram community (DE)
                </h3>
                <p>
                  t.me
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <GoMarkGithub />
              </div>
              <div>
                <h3>
                  Report an issue
                </h3>
                <p>
                  t.me
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <MdHelp />
              </div>
              <div>
                <h3>
                  Frequently-Asked Questions
                </h3>
                <p>
                  defichain.com
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <Logo />
              </div>
              <div>
                <h3>
                  Official website
                </h3>
                <p>
                  defichain.com
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <FaReddit />
              </div>
              <div>
                <h3>
                  Reddit community
                </h3>
                <p>
                  reddit.com
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
        <Col
          md="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            onClick={[Function]}
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <div>
                <MdBook />
              </div>
              <div>
                <h3>
                  Community Wiki
                </h3>
                <p>
                  defichain-wiki.com
                </p>
              </div>
            </CardBody>
          </Card>
        </Col>
      </Row>
      <p>
        DeFiChain is a community-driven and open project.
        <br />
        The DeFiChain Foundation does not provide direct support.
      </p>
    </section>
  </div>
</div>
`;
